server/controllers/index.js

```diff
@@ -80,22 +102,26 @@ router.get('/api/start', auth, (req, res) => {
 router.get('/api/end', auth, (req, res) => {
     const { roomId } = req.session;
     try {
-        const gameState = Room.getById(req.session.roomId).gameState;
-        gameState.endGame();
-        Room.deleteById(roomId);
-        closeRoom(roomId);
-        res.sendStatus(200);
+        const gameState = Room.getById(roomId).gameState;
+        db.updateScores(gameState.getPlayers()).then((user) => {
+            gameState.endGame();
+            Room.deleteById(roomId);
+            closeRoom(roomId);
+            res.sendStatus(200);
+        });
     } catch (err) {
         res.status(400).json({message: err.message});
     }
 });
 
 /* Current player plays a card and a word */
 router.post('/api/play-card-word', auth, (req, res) => {
-    if (Room.getById(req.session.roomId).gameState.isCurrentPlayer(req.session.user)) { /* Only current player is allowed to play both a word and a card */
+    const { user, roomId } = req.session;
+    const { word, cardId } = req.body;
+    if (Room.getById(roomId).gameState.isCurrentPlayer(user.username)) { /* Only current player is allowed to play both a word and a card */
```

Comment:
We should probably move this validation into the GameLogic itself, seeing as we're doing try catch, same in the play-card request

Reply:
sure, should I do it in this PR?